src/main/java/com/algorand/algosdk/transaction/SignedTransaction.java

```diff
@@ -19,23 +20,32 @@
     public Signature sig = new Signature(); // can be null
     @JsonProperty("msig")
     public MultisigSignature mSig = new MultisigSignature();
+    @JsonProperty("lsig")
+    public LogicsigSignature lSig = new LogicsigSignature();
 
     @JsonIgnore
     public String transactionID = "";
 
-    public SignedTransaction(Transaction tx, Signature sig, MultisigSignature mSig, String transactionID) {
+    public SignedTransaction(
+        Transaction tx, Signature sig, MultisigSignature mSig, LogicsigSignature lSig, String transactionID
+    ) {
         this.tx = Objects.requireNonNull(tx, "tx must not be null");
         this.mSig = Objects.requireNonNull(mSig, "mSig must not be null");
         this.sig = Objects.requireNonNull(sig, "sig must not be null");
+        this.lSig = Objects.requireNonNull(lSig, "lSig must not be null");
         this.transactionID = Objects.requireNonNull(transactionID, "txID must not be null");
     }
 
     public SignedTransaction(Transaction tx, Signature sig, String txId) {
-        this(tx, sig, new MultisigSignature(), txId);
+        this(tx, sig, new MultisigSignature(), new LogicsigSignature(), txId);
```

Comment:
Why is it called twice? I agree but with a delegated call we have only one place to check input params instead of four. We might want to change but possible as another PR.